From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 1 Feb 2016 11:28:00 +0100
Subject: [PATCH] {BUG} drm: gma500/mdfld_dsi_output: fix
 mdfld_dsi_connector_dpms signature
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc reports:

    drivers/gpu/drm/gma500/mdfld_dsi_output.c:407:39: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
      .dpms = /*drm_helper_connector_dpms*/mdfld_dsi_connector_dpms,
                                           ^
    drivers/gpu/drm/gma500/mdfld_dsi_output.c:407:39: note: (near initialization for ‘mdfld_dsi_connector_funcs.dpms’)
    cc1: all warnings being treated as errors
---
 drivers/gpu/drm/gma500/mdfld_dsi_output.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/gma500/mdfld_dsi_output.c b/drivers/gpu/drm/gma500/mdfld_dsi_output.c
index d758f4cc6805..b1f2ce896848 100644
--- a/drivers/gpu/drm/gma500/mdfld_dsi_output.c
+++ b/drivers/gpu/drm/gma500/mdfld_dsi_output.c
@@ -382,14 +382,14 @@ static int mdfld_dsi_connector_mode_valid(struct drm_connector *connector,
 	return MODE_OK;
 }
 
-static void mdfld_dsi_connector_dpms(struct drm_connector *connector, int mode)
+static int mdfld_dsi_connector_dpms(struct drm_connector *connector, int mode)
 {
 	if (mode == connector->dpms)
-		return;
+		return -1 /* TODO: which return value? */;
 
 	/*first, execute dpms*/
 
-	drm_helper_connector_dpms(connector, mode);
+	return drm_helper_connector_dpms(connector, mode);
 }
 
 static struct drm_encoder *mdfld_dsi_connector_best_encoder(
-- 
